src/submission/models.py

```diff
@@ -1154,6 +1154,22 @@ def funders(self):
         """Method replaces the funders m2m model for backwards compat."""
         return ArticleFunding.objects.filter(article=self)
 
+    @property
+    def date_archived(self):
+        # Return early if the article is not actually archived.
+        if self.stage != STAGE_ARCHIVED:
+            return None
+
+        log_entry = ArticleStageLog.objects.filter(
+            article=self,
+            stage_to='Archived',
+        ).first()
+
+        if log_entry:
+            return log_entry.date_time
+
+        return None
+
```

Comment:
Is it worth also annotating this date to the new `rejected_and_archived_articles` query to avoid another `n` times roundtrip?

Reply:
Yeah - not sure why it didn't occur to me.